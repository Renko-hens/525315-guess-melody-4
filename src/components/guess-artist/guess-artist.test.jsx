import React from "react";
import renderer from "react-test-renderer";
import GuessArtistScreen from "./guess-artist.jsx";

const AVATAR_URL = `https://api.adorable.io/avatars/128`;

const questionArtist = {
  type: `artist`,
  song: {
    artist: `Jim Beam`,
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
  },
  answers: [
    {
      picture: `${AVATAR_URL}}`,
      artist: `Keanu Reves`,
    },
    {
      picture: `${AVATAR_URL}}`,
      artist: `John Wick`,
    },
    {
      picture: `${AVATAR_URL}}`,
      artist: `Neo Man`,
    },
  ],
};

describe(`Render Artist Screen components`, () => {
  it(`Render Artist Screen`, () => {
    const tree = renderer.create((
      <GuessArtistScreen
        onAnswer={() => {}}
        question={questionArtist}
      />), {
      createNodeMock: () => {
        return {};
      }
    }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});


