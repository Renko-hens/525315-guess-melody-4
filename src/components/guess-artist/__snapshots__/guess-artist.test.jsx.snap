// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Artist Screen components Render Artist Screen 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <div
      className="track"
    />
  </div>
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-0"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-0"
      />
      <label
        className="artist__name"
        htmlFor="answer-0"
      >
        <img
          alt="Keanu Reves"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128}"
        />
        Keanu Reves
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-1"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="John Wick"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128}"
        />
        John Wick
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-2"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-2"
      />
      <label
        className="artist__name"
        htmlFor="answer-2"
      >
        <img
          alt="Neo Man"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128}"
        />
        Neo Man
      </label>
    </div>
  </form>
</section>
`;
